bayes_opt/acquisition.py

```diff
@@ -0,0 +1,854 @@
+"""Acquisition functions for Bayesian Optimization."""
+import warnings
+import numpy as np
+from numpy.random import RandomState
+from scipy.optimize import minimize
+from scipy.stats import norm
+from scipy.special import softmax
+from .target_space import TargetSpace
+from .constraint import ConstraintModel
+from sklearn.gaussian_process import GaussianProcessRegressor
+from typing import Callable, List, Union, Tuple
+from copy import deepcopy
+from numbers import Number
+
+class ConstraintNotSupportedError(Exception):
+    """Raised when constrained optimization is not supported."""
+
+    pass
+
+
+class NoValidPointRegisteredError(Exception):
+    """Raised when an acquisition function depends on previous points but none are registered."""
+
+    pass
+
+
+class TargetSpaceEmptyError(Exception):
+    """Raised when the target space is empty."""
+
+    pass
+
+
+class AcquisitionFunction():
```

Comment:
could use the [ABC](https://docs.python.org/3/library/abc.html) functionality to define the abstract base class, as well as any abstract methods using `@abstractmethod`

Reply:
good call, I changed it as you suggested